include/avif/avif.h

```diff
@@ -552,6 +564,14 @@ typedef enum avifChromaUpsampling
     AVIF_CHROMA_UPSAMPLING_BILINEAR = 4      // Uses bilinear filter (built-in)
 } avifChromaUpsampling;
 
+typedef enum avifChromaDownsampling
+{
+    AVIF_CHROMA_DOWNSAMPLING_AUTOMATIC = 0,    // Chooses best trade off of speed/quality (prefers libyuv, else uses BEST_QUALITY)
+    AVIF_CHROMA_DOWNSAMPLING_FASTEST = 1,      // Chooses speed over quality (prefers libyuv, else uses AVERAGE)
+    AVIF_CHROMA_DOWNSAMPLING_BEST_QUALITY = 2, // Chooses the best quality upsampling (avoids libyuv, uses AVERAGE)
+    AVIF_CHROMA_DOWNSAMPLING_FLOAT_AVERAGE = 3 // Uses floating point RGB-to-YUV conversion then averaging (built-in)
```

Comment:
Let's remove "FLOAT_" from this enum name.

I discussed the API problem with Joe Drago and he suggested that we add a `preferLibYUV` setting to control whether libyuv should be used if available. Then the `chromaUpsampling` and `chromaDownsampling` settings will only specify the upsampling and downsampling filters. To prepare for this future change, this enum name just needs to specify that chroma is downsampled by averaging.

Reply:
So should we keep `chromaUpsampling` and `chromaDownsampling` in `avifRGBImage` or should we move them to `vifImageRGBToYUV()` and `avifImageYUVToRGB()` arguments as discussed? I guess the `preferLibYUV` setting will be at the same location.

Reply:
In this pull request we will keep `chromaUpsampling` and `chromaDownsampling` in `avifRGBImage`, as you suggested earlier.

Reply:
Yes. I meant in the future, not in this PR. Sorry, that was unclear.

Reply:
No worries.

We will need to think about this API design issue. Ideally these settings should become function parameters for `avifImageRGBToYUV()` and `avifImageYUVToRGB()`. But we need to avoid changing the function signatures again if we ever need to add another new setting. So we may need to use a bitmask (and combine these settings using bitwise-OR) or a parameter struct.

Reply:
I sent https://github.com/AOMediaCodec/libavif/pull/1026 with modifications to `avif.h` only, as a start. @wantehchang could you take a quick look and tell me if it matches what Joe and you discussed, please? I will refactor the remaining source files afterwards, to avoid unnecessary work in the wrong direction.